Sources/LLVM/Module.swift

```diff
@@ -123,11 +123,64 @@ public final class Module {
     }
   }
 
+  /// Retrieves the sequence of functions that make up this module.
+  public var functions: AnySequence<Function> {
+    var current = firstFunction
+    return AnySequence<Function> {
+      return AnyIterator<Function> {
+        defer { current = current?.next() }
+        return current
+      }
+    }
+  }
+
+  /// Retrieves the first function in this module, if there are any functions.
+  public var firstFunction: Function? {
+    guard let fn = LLVMGetFirstFunction(llvm) else { return nil }
+    return Function(llvm: fn)
+  }
+
+  /// Retrieves the last function in this module, if there are any functions.
+  public var lastFunction: Function? {
+    guard let fn = LLVMGetLastFunction(llvm) else { return nil }
+    return Function(llvm: fn)
+  }
+
+  /// Retrieves the first global in this module, if there are any globals.
+  public var firstGlobal: Global? {
+    guard let fn = LLVMGetFirstGlobal(llvm) else { return nil }
+    return Global(llvm: fn)
+  }
+
+  /// Retrieves the last global in this module, if there are any globals.
+  public var lastGlobal: Global? {
+    guard let fn = LLVMGetLastGlobal(llvm) else { return nil }
+    return Global(llvm: fn)
+  }
+
+  /// Retrieves the sequence of functions that make up this module.
+  public var globals: AnySequence<Global> {
+    var current = firstGlobal
+    return AnySequence<Global> {
+      return AnyIterator<Global> {
+        defer { current = current?.next() }
+        return current
+      }
+    }
+  }
+
   /// Dump a representation of this module to stderr.
   public func dump() {
     LLVMDumpModule(llvm)
   }
 
+  /// The full text IR of this module
+  public var description: String {
+    let cStr = LLVMPrintModuleToString(llvm)!
```

Comment:
I can't think of any other thing for `description`.
This would also allow for `print(module)` to print the IR.